### PR TITLE
Add R script to get data from admiral templates

### PR DESCRIPTION
**Pull request by dgrassellyb**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately 
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by zdz2101 — APPROVED** [review]

**Comment by dgrassellyb** [comment]
Hi @manciniedoardo / @kaz462 / @zdz2101 ! I am going on vacation this evening, and coming back the 2nd of January. if you want to update the labels you can still update the file `/inst/extdata/adams-specs.xlsx` on this branch, and rerun the script `data-raw/create_adams_data.R` to update the data !

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
-Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   
+Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles//development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.
```
Comment (bundfussr) [inline .github/pull_request_template.md:1]:
```suggestion
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
+- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles//unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
+- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles//programming_strategy.html#deprecation)?
+- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles//programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
```
Comment (bundfussr) [inline .github/pull_request_template.md:9]:
```suggestion
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
```

File: data-raw/create_adams_data.R [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    "#'",
+    sprintf("#' @name %s", dataset_name),
+    "#' @docType data",
+    sprintf("#' @format A data frame with %s columns:", ncol(data)),
+    "#'   \\describe{",
+    paste(sapply(names(data), function(col_name) {
+      paste(sprintf("#'     \\item{ %s }{%s}", col_name, col_name))
+    }, USE.NAMES = FALSE), collapse = "\n"),
+    "#'   }",
+    "#'", sprintf("#' @source Generated from create_adams_data.R script."),
```
Comment (bundfussr) [inline data-raw/create_adams_data.R:38]: I would replace `create_adams_data.R` with the name of the template file and the package. I think this is more helpful for the users.

Comment (dgrassellyb) [reply data-raw/create_adams_data.R:38]: I think this is a good idea ! But maybe I would create a separated issue for this (I think @manciniedoardo and @kaz462 want to release pharmaverseadam early January)

**Review by kaz462 — CHANGES_REQUESTED** [review]

File: data-raw/create_adams_data.R [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    sprintf("#' Dataset %s", dataset_name),
+    "#'",
+    sprintf("#' %s dataset", dataset_name),
+    "#'",
+    sprintf("#' @name %s", dataset_name),
+    "#' @docType data",
+    sprintf("#' @format A data frame with %s columns:", ncol(data)),
+    "#'   \\describe{",
+    paste(sapply(names(data), function(col_name) {
+      paste(sprintf("#'     \\item{ %s }{%s}", col_name, col_name))
```
[the remark this answers is not in the document]
Comment (dgrassellyb) [reply data-raw/create_adams_data.R:35]: Yes good point it's something we discussed with @manciniedoardo (I think I can embed this with Stefen suggestion (https://github.com/pharmaverse/pharmaverseadam/pull/53#discussion_r1433947662) )

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply data-raw/create_adams_data.R:35]: @kaz462 @dgrassellyb the labels have all been compiled in [the shared gsheet](url). is it possible to now download this sheet as an excel file and upload it to the package in this PR so that we can both populate the datasets with labels and make sure the labels appear on the website with this PR? Or are there added dificulties?

File: R/adbcva_admiralophtha.R
[the remark this answers is not in the document]
Comment (dgrassellyb) [reply R/adbcva_admiralophtha.R:1]: Mmh good question .. the data folder contains the adams datasets produced by the templates and the files `R/adoe_admiralophtha.R`, `R/adrs_admiralonco.R` are used for the documentation. I think it depends how you want to proceed to maintain this package ? (I thought the idea was to call `create_adams_data.R` and push changes in a new branch and submit a PR (then another dev can look at the changes before approving the PR?) - but for sure a git diff cmd won't help a lot on .rda files (my advise is to use `diffdf` package to display differences)

File: data-raw/create_adams_data.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (dgrassellyb) [reply data-raw/create_adams_data.R:35]: Hi @manciniedoardo I updated the script (but didn't push the updates yet, need to test) - but I can not access the gsheet (your link is not working) - could you please send me the gsheet to be able to get latest labels ?

[the remark this answers is not in the document]
Comment (manciniedoardo) [reply data-raw/create_adams_data.R:35]: Hi @dgrassellyb apologies, shared with you now.

**Comment by dgrassellyb** [comment]
@manciniedoardo : since release 1.0.0 I am having some issues.. I noticed there are now dependencies with internal packages data (admiral - admiral_adsl for example) - I can not access the data information on the imported package (I guess when we submit to CRAN we ignore data folder?) - this is quite a major issue and it would be great to be more aware of such changes for pharmaverseadam !

**Comment by dgrassellyb** [comment]
> @manciniedoardo : since release 1.0.0 I am having some issues.. I noticed there are now dependencies with internal packages data (admiral - admiral_adsl for example) - I can not access the data information on the imported package (I guess when we submit to CRAN we ignore data folder?) - this is quite a major issue and it would be great to be more aware of such changes for pharmaverseadam !

(mmh just found a way to load the data, but I will need more time to make the updates with this change)

**Review by manciniedoardo — APPROVED** [review]

File: R/adbcva_admiralophtha.R
[the remark this answers is not in the document]
Comment (kaz462) [reply R/adbcva_admiralophtha.R:1]: > I thought the idea was to call create_adams_data.R and push changes in a new branch and submit a PR (then another dev can look at the changes before approving the PR?

agree. 
I meant to remove `adoe_admiralophtha.R/rda` since we have `adoe_ophtha.R/rda`, it seems `xx_admiralxx.R/rda` files are removed - all good now.  @dgrassellyb Thanks!

File: inst/extdata/~$adams-specs.xlsx
Comment (kaz462) [inline inst/extdata/~$adams-specs.xlsx:1]: can we remove this file?

**Review by kaz462 — CHANGES_REQUESTED** [review]
@dgrassellyb thanks for the udpates. Labels are not applied to datasets from `admiralvaccine`, all other ones look great!

image

**Comment by dgrassellyb** [comment]
> @dgrassellyb thanks for the udpates. Labels are not applied to datasets from `admiralvaccine`, all other ones look great!
> 
> image

It's due to some internal errors on admiralvaccine - I could not run these templates (@manciniedoardo told me some fixes should be pushed to admiralvaccine repo, once it's done, we can re-run the script to update vaccine datasets and doc !) - Do you think we can still proceed with the release or should we wait for vaccine to be ready ?

**Review by manciniedoardo — APPROVED** [review]
